From: Donald Hunter <donald.hunter@gmail.com>
Date: Sat, 27 May 2023 14:31:05 +0100
Subject: tools: ynl: Initialise fixed headers to 0 in genetlink-legacy
Patch-mainline: v6.5-rc1
Git-commit: 5ac18889bde04ed2d4f2559da01e9b160234525c
References: jsc#PED-3311

This eliminates the need for e.g. --json '{"dp-ifindex":0}' which is not
too big a deal for ovs but will get tiresome for fixed header structs that
have many members.

Signed-off-by: Donald Hunter <donald.hunter@gmail.com>
Signed-off-by: Jakub Kicinski <kuba@kernel.org>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 tools/net/ynl/lib/ynl.py |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/tools/net/ynl/lib/ynl.py
+++ b/tools/net/ynl/lib/ynl.py
@@ -541,7 +541,7 @@ class YnlFamily(SpecFamily):
         if op.fixed_header:
             fixed_header_members = self.consts[op.fixed_header].members
             for m in fixed_header_members:
-                value = vals.pop(m.name)
+                value = vals.pop(m.name) if m.name in vals else 0
                 format = NlAttr.get_format(m.type, m.byte_order)
                 msg += format.pack(value)
         for name, value in vals.items():
